client.go

```diff
@@ -327,18 +327,18 @@ func (c *Client) WaitForTask(task *Task, options ...waitParams) (*Task, error) {
 // ApiKey options is the API key parent of the token. If you leave it empty the client API Key will be used.
 func (c *Client) GenerateTenantToken(SearchRules interface{}, Options *TenantTokenOptions) (resp string, err error) {
 	// Validate the arguments
-	if (SearchRules == nil) {
+	if SearchRules == nil {
```

Comment:
Is it not possible in this case to check for these types? Because if `searchRules` is a integer it will not enter the if condition.
Same for the other if conditions

Reply:
I change the type of `searchRules`. What do you mean by the other if conditions?

Reply:
Is it not useful to check that `searchRules` is not an integer for example? Why do we only check if it is a nil ?